pytest/tests/mocknet/mirror.py

```diff
@@ -414,21 +422,18 @@ def start_traffic_cmd(args, traffic_generator, nodes):
 
 
 def update_binaries_cmd(args, traffic_generator, nodes):
-    pmap(lambda node: node.neard_runner_update_binaries(),
-         nodes + [traffic_generator])
+    if (args.neard_binary_url or args.update_epoch_height
+       ) and not (args.neard_binary_url and args.update_epoch_height):
+        logger.warning(f'specify both neard_binary_url and update_epoch_height')
+        return
+    pmap(
+        lambda node: node.neard_runner_update_binaries(
+            args.neard_binary_url, args.update_epoch_height),
+        nodes + to_list(traffic_generator))
```

Comment:
can we also add dumper node to this list? as it is now separate from traffic generator?

Reply:
the dumper node is included in the nodes list. At a later time, I may refactor this nodes list to be a class:
```
class Nodes:
  state_dumpers: [Node]
  validators: [Node]
  rpc: [Node]
  traffic: Node

  def get_nodes(filters):
   # filter nodes
   return [traffic] + rpc + validators + state_dumpers
```